drivers/adodb-firebird.inc.php

```diff
@@ -255,37 +355,65 @@ function &MetaIndexes ($table, $primary = FALSE, $owner=false)
 	}
 
 
-	// See http://community.borland.com/article/0,1410,25844,00.html
-	function RowLock($tables,$where,$col=false)
-	{
-		if ($this->autoCommit) {
-			$this->BeginTrans();
-		}
-		$this->Execute("UPDATE $table SET $col=$col WHERE $where "); // is this correct - jlim?
-		return 1;
-	}
-
-
-	function CreateSequence($seqname = 'adodbseq', $startID = 1)
+	 // See http://community.borland.com/article/0,1410,25844,00.html
+	/**
+	 * Lock a table row for a duration of a transaction.
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:rowlock
+	 * @link https://firebirdsql.org/refdocs/langrefupd21-notes-withlock.html
+	 *
+	 * @param string $tables The table(s) to lock rows for.
+	 * @param string $where (Optional) The WHERE clause to use to determine which rows to lock.
+	 * @param string $col (Optional) The columns to select.
+	 *
+	 * @return bool True if the locking SQL statement executed successfully, otherwise false.
+	 */
+	public function rowLock($tables,$where,$col=false)
 	{
-		$ok = $this->Execute(("CREATE GENERATOR $seqname" ));
-		if (!$ok) return false;
-		return $this->Execute("SET GENERATOR $seqname TO ".($startID-1));
+		if ($this->transCnt==0) 
+			$this->beginTrans();
+		
+		if ($where) $where = ' where '.$where;
+		$rs = $this->execute("SELECT $col FROM $tables $where FOR UPDATE WITH LOCK");
+		return 
+			!empty($rs);
 	}
 
-	function DropSequence($seqname = 'adodbseq')
+	/**
+	 * Creates a sequence in the database.
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:createsequence
+	 *
+	 * @param string $seqname The sequence name.
+	 * @param int $startID The start id.
+	 *
+	 * @return ADORecordSet|bool A record set if executed successfully, otherwise false.
+	 */
+	public function createSequence($seqname='adodbseq', $startID = 1)
 	{
-		$seqname = strtoupper($seqname);
-		return $this->Execute("DROP GENERATOR $seqname");
-	}
-
-	function GenID($seqname='adodbseq',$startID=1)
+		$sql = sprintf($this->_genSeqSQL,$seqname,$startID);
+		$ok = $this->Execute($sql);
+		
+		if (!$ok) 
+			return false;
+	} 
+	
+	/**
+	 * A portable method of creating sequence numbers.
+	 *
+	 * @link https://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:genid
+	 *
+	 * @param string $seqname (Optional) The name of the sequence to use.
+	 * @param int $startID (Optional) The point to start at in the sequence.
+	 *
+	 * @return bool|int|string
```

Comment:
Looks like we're only returning _int_ here

Reply:
Updated doc